subsys/bluetooth/host/Kconfig

```diff
@@ -47,7 +47,7 @@ config BT_HCI_TX_STACK_SIZE
 	prompt "HCI Tx thread stack size" if BT_HCI_TX_STACK_SIZE_WITH_PROMPT
 	default 512 if BT_H4
 	default 512 if BT_H5
-	default 416 if BT_SPI
+	default 1024 if BT_SPI
```

Comment:
This looks like a significant increase in stack size, any idea on the root cause?
Is it possible that application was compiled in debug mode or with some logging enabled?

Reply:
I didn't really measure it. I can try to, yeah. This is probably due to some update to the nordic hal that requires more stack.

In general it's probably a bad idea to have a hand-tuned TX stack size, as the TX stack calls directly into the driver and the stack depth required by the vendor hal can vary a lot.
 
I would rather prefer comfortable defaults, like the rest of the BT stacks, and the application developer can do the fine-tuning. Nothing worse IMO than booting a sample and being greeted by the panic handler.

Reply:
> with some logging enabled

If we enable logging and it makes the application stack-overflow then it's a bad default IMO.

Reply:
I agree, but I experienced this in the past. I am not sure what is the recommended way to define thread's stack size.
Normally, assuming no recursive functions are present, it should be the maximum of each possible calling tree.
If no function pointers are present, I guess the compiler could do the job.
In general, it's a difficult task, but increasing all the threads stack to be on the safe side could lead to significant waste of memory.

I know also that in shell, you can inspect stack usage at runtime.

Is there a guideline in Zephyr on how to define stack size for threads?

Reply:
> If no function pointers are present

Doesn't zephyr use this a whole lot? at least for HW drivers definitely. I think also for HCI drivers.
I asked during today's zephyr host meeting and didn't get concrete answers, so I agree a line in the doc would be great.